ci/poll-extension-availability.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+# Query the installed extension until the expected version is available.
+# Inspired from: https://github.com/microsoft/azure-devops-extension-tasks/issues/189#issue-820340124
+
+set -e
+
+SERVICE_URL=
+SERVICE_TOKEN=
+TASK_ID=
+TASK_VERSION=
+
+while getopts t:s:c:v: flag; do
+    case "${flag}" in
+        s) SERVICE_URL="${OPTARG}";;
+        t) SERVICE_TOKEN="${OPTARG}";;
+        c) TASK_ID="${OPTARG}";;
+        v) TASK_VERSION="${OPTARG}";;
+    esac
+done
+
+attempts=0
+max=25
+
+until [ $attempts -gt $max ];
+do
+    extensions=$(tfx build tasks list --authType pat --service-url $SERVICE_URL -t $SERVICE_TOKEN --no-color --json)
+    installed_version=$(echo $extensions | jq -r ".[] | select(.contributionIdentifier == \"$TASK_ID\") | .contributionVersion")
+
+    if [ -z $installed_version ]
+    then
+        echo "No version found for $TASK_ID"
+        exit 1
+    fi
```

Comment:
I'm not sure to understand why we need this condition. If there isn't any version found, it might be the very first is not there yet? I guess there is another edge-case covered here, but can't find it.

Reply:
The edge case is if we target the wrong `--service-url`, or the wrong task id, or that the JSON result is empty for another reason.

It's just a fail safe to ensure we don't poll for a long time (1 + 2 + ... + 25 = 325 seconds ~ 5 minutes) for nothing if we configure the script improperly.